rpc/client/client_test.go

```diff
@@ -499,7 +500,76 @@ func TestConsensusState(t *testing.T) {
 }
 
 func TestBlockchainInfo(t *testing.T) {
-	panic("todo")
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc := getRPC(t)
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+
+	heights := []int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	for _, h := range heights {
+		block := getRandomBlock(uint64(h), 5)
+		err := rpc.node.Store.SaveBlock(block, &types.Commit{
+			Height:     uint64(h),
+			HeaderHash: block.Header.Hash(),
+		})
+		require.NoError(err)
+	}
+
+	tests := []struct {
+		desc string
+		min  int64
+		max  int64
+		exp  []*tmtypes.BlockMeta
+		err  bool
+	}{
+		{
+			desc: "min = 1 and max = 5",
+			min:  1,
+			max:  5,
+			exp:  []*tmtypes.BlockMeta{getBlockMeta(rpc, 1), getBlockMeta(rpc, 5)},
```

Comment:
This doesn't test everything. As you're testing that `BlockMeta` generated with `LoadBlock` and `ToABCIBlockMeta` is the same as... `BlockMeta` generated with `LoadBlock` and `ToABCIBlockMeta` ;)
This way we're not able to catch any issues with `ToABCIBlockMeta`.
You should change this test (provide manually filled `BlockMeta`), or add a unit test in `conv` package to ensure that `ToABCIBlockMeta` works fine.
Or both.

Reply:
Tbh this did occur to me while writing the `ToABCIBlockMeta`. But since there were no tests for the abci conversions, I decided to focus just on testing the rpc method (and leave the `ToABCIBlockMeta` test  for a  seperate PR).